### ISSUE TITLE
`symbol not found` error using MacOS Sonoma

### ISSUE THREAD

**Issue by fuetser** [AstroNvim/AstroNvim#2311]
### Checklist

- [X] Using a stable version of Neovim (i.e. not neovim nightly)
- [X] `:AstroUpdate`
- [X] Restarted AstroNvim

### Operating system/version

macOS Sonoma 14.0

### Terminal/GUI

iterm2

### AstroNvim Health

AstroNvim ~
- AstroNvim Version: v3.37.0
- Neovim Version: v0.9.2
- OK Using stable Neovim >= 0.8.0
- OK `git` is installed: Used for core functionality such as updater and plugin management
- OK `open` is installed: Used for `gx` mapping for opening files with system opener (Optional)
- OK `lazygit` is installed: Used for mappings to pull up git TUI (Optional)
- OK `node` is installed: Used for mappings to pull up node REPL (Optional)
- WARNING `gdu` is not installed: Used for mappings to pull up disk usage analyzer (Optional)
- WARNING `btm` is not installed: Used for mappings to pull up system monitor (Optional)
- OK `python3` is installed: Used for mappings to pull up python REPL (Optional)

### Describe the bug

I installed the latest version of neovim then installed Astro using git clone and opened neovim. Right after the plugins finished installing i got this error. If i try to open file explorer or open a file i get the same error message.  I also did all of this on a new computer which did not have iterm2, brew or neovim installed before. I have tried reinstalling neovim and Astro but nothing helped. 

### Steps to Reproduce

1. brew install neovim
2. git clone --depth 1 https://github.com/AstroNvim/AstroNvim ~/.config/nvim
3. nvim

### Expected behavior

Plugins are successfully installed and no error messages are shown. 

### Screenshots

<img width="1512" alt="Снимок экрана 2023-09-30 в 17 46 15" src="https://github.com/AstroNvim/AstroNvim/assets/51047596/655866f4-2c2e-458c-a2ac-77619fe6b191">

**Comment by aravezskinteeth** [comment]
I can confirm that I'm getting the same error as you, even after I cloned my AstroNvim config. I have no idea how to solve it.

**Comment by andrewozh** [comment]
Same thing! I upgraded to Sonoma, also upgraded all brew packages
Complete AstroNvim reinstall didn't help

**Comment by andrewozh** [comment]
@fuetser I found out that the problem is caused by some folding mechanism from new version of nvim, and we supposed to have ability to gracefully disable it by adding    
```
foldenable = false,
foldcolumn = nil,
```
to `~/.config/nvim/lua/user/options.lua`, but it won't help for some reason..
So I changed it directly in `~/.config/nvim/lua/astronvim/options.lua`
After this the amount of errors decreased significantly!
At least nvim is slightly usable with this, and I can work tomorrow 🎉

But I still see same errors from `nvim-ufo` plugin! I tried to disable it in my user configuration by adding `~/.config/nvim/lua/user/plugins/ufo.lua`:
```
return {
  {
    "kevinhwang91/nvim-ufo",
    enable = false,
    dependencies = { "kevinhwang91/promise-async" },
    provider_selector = function(_, filetype, buftype)
      -- force do nothing
    end
  },
}
```

But it didn't help as well.. I did evetything, set enable to false, deleted events, cleared all actions from it's function, and it still run somehow


@mehalter userconf options and plugins seems not work for me, or am I doing sth wrong 😔

**Comment by anuragrao04** [comment]
> Same thing! I upgraded to Sonoma, also upgraded all brew packages Complete AstroNvim reinstall didn't help

Same steps followed, same issue +1

**Comment by fuetser** [comment]
I manually installed neovim v0.8.3 and that error doesn't show up anymore. It does't solve the original issue but is ok as a workaround.

**Comment by mehalter** [comment]
I looked into this and it looks like this is related to using pre-release versions of Mac OS which is known to break FFI. It looks like probably just needs to use nightly neovim or something or use a stable operating system. Sorry I'm not much help! 

https://github.com/neovim/neovim/issues/25295

**Comment by mehalter** [comment]
https://github.com/neovim/neovim/pull/25274

Looks like this PR probably fixes this for Mac OS Sonoma

Also I looked into this and I believe the mention of it being a pre-release is not true anymore since it is fully released. I believe it just needed to be fixed in Neovim to support the new OS. Hopefully they release some patch updates for neovim that backport this fix or something!

**Comment by aravezskinteeth** [comment]
> [neovim/neovim#25274](https://github.com/neovim/neovim/pull/25274)
> 
> Looks like this PR probably fixes this for Mac OS Sonoma
> 
> Also I looked into this and I believe the mention of it being a pre-release is not true anymore since it is fully released. I believe it just needed to be fixed in Neovim to support the new OS. Hopefully they release some patch updates for neovim that backport this fix or something!

Downloaded the `--HEAD` version of `neovim` with `brew` and the errors seem to have disappeared, thankfully.

**Comment by mehalter** [comment]
This has been fixed in Neovim 0.9.4.
